python/tskit/drawing.py

```diff
@@ -791,82 +795,96 @@ def draw_x_axis(
                 transform="translate(0 -11)",
                 text_anchor="middle",
             )
-        if self.x_axis:
-            if tick_length_upper is None:
-                tick_length_upper = tick_length_lower
-            y = rnd(self.plotbox.max_y - self.x_axis_offset)
-            dash_locs = np.concatenate(
-                (
-                    [self.plotbox.left],
-                    self.x_transform(alternate_dash_positions),
-                    [self.plotbox.right],
+        if not self.x_axis:
+            return
+        if len(x_regions) > 0:
+            regions_group = x_axis.add(dwg.g(class_="x-regions"))
+            for i, ((left, right), label) in enumerate(x_regions.items()):
```

Comment:
I think you need some error checking on the region tuples here  - just 
```0 <= left < right <= ts.sequence_length`` is fine.

Reply:
Yep, that seems sensible. I'll add